biogeochem/EDMortalityFunctionsMod.F90

```diff
@@ -75,12 +82,41 @@ subroutine mortality_rates( cohort_in,bc_in,cmort,hmort,bmort,frmort )
     logical, parameter :: test_zero_mortality = .false. ! Developer test which
                                                         ! may help to debug carbon imbalances
                                                         ! and the like
+     
+   ! Size Dependent Senescence
+    ! rate (r) and inflection point (ip) define the increase in mortality rate with dbh
+    mort_r_size_senescence = EDPftvarcon_inst%mort_r_size_senescence(cohort_in%pft)
+    mort_ip_size_senescence = EDPftvarcon_inst%mort_ip_size_senescence(cohort_in%pft)
+    
+    ! if param values have been set then calculate smort
+    if ( mort_ip_size_senescence < check_initialized ) then 
+       smort = 1.0_r8 / ( 1.0_r8 + exp( -1.0_r8 * mort_r_size_senescence * &
+            (cohort_in%dbh - mort_ip_size_senescence) ) ) 
+    else
+       smort = 0.0_r8
+    end if
+
+    ! if user has set cohort age fusion param to not be _ we calculate age
```

Comment:
What is _ in this sentence? Also why doesn't this use 'cohort_age_tracking'?

Reply:
I've changed it to match the comment above for size mortality. The size and age dependent mortality parameters are _ by default in the parameter file. The user has to set them in order to turn on either mortality term. Cohort age tracking is turned on separately with the cohort_age_tracking parameter. This was in case anyone in the future wanted to track cohort age independently of having size or age mortality turned on.

Reply:
check_initialized is defined in PRTGenericMod.F90, but that check is designed to work in step with un_initialized.  See here:

https://github.com/NGEET/fates/blob/master/parteh/PRTGenericMod.F90#L43

It only works in situations where we are defining uninitialized values with "un_initialized".  In this case, the parameter values that have "_"  from the file, are not the same as un_initialized.  

I remember we thought it was 9.9E36, but there was some confusion if we were importing that value correctly...  I had thought that "spval" from CLM/ELM would give us this value.

We have an "ignore" value for history, but that value is not necessarily the "no data" value we get from netcdf:  https://github.com/NGEET/fates/blob/master/main/FatesInterfaceMod.F90#L94

Reply:
Ah ok, thank you.  I misunderstood the un_initialized value. Would it make most sense then for me to define a local arbitrary large number to compare these parameters against, in order to check whether or not they have been set by the user in the parameter file?

Reply:
I think the fill values we were printing out when they made there way to FATES were 1E36 right? If so, I think this is is where it is defined...

https://github.com/ESCOMP/CTSM/blob/master/src/main/ncdio_pio.F90.in#L146

Reply:
If I print a parameter that is "_" in the parameter file I get 9.9692099...E+036.

Reply:
I think it is defined in ./cime/src/externals/pio2/src/flib/pio_types.F90 L234-235

Reply:
I've made a new variable called 'fates_check_param_set' defined in FatesConstantsMod to check whether the mortality parameters have been set. This might not be the way to go so happy to change it if there is a better way of doing things. But it seems to be working.

Reply:
@rgknox I think the discussion from yesterday is here.